servers/rendering_server.cpp

```diff
@@ -1239,6 +1239,13 @@ Error RenderingServer::mesh_create_surface_data_from_arrays(SurfaceData *r_surfa
 		// Include custom array format type.
 		if (format & (1ULL << (ARRAY_CUSTOM0 + i))) {
 			format |= (RS::ARRAY_FORMAT_CUSTOM_MASK << (RS::ARRAY_FORMAT_CUSTOM_BASE + i * RS::ARRAY_FORMAT_CUSTOM_BITS)) & p_compress_format;
+
+			// If the mesh contains no vertex array, infer the array length from the custom array.
+			if (array_len == 0) {
+				Vector<Variant> custom_array = p_arrays[RS::ARRAY_CUSTOM0 + i];
+				uint32_t factor_reciprocal = _get_vertex_to_custom_array_length_factor(format, RS::ARRAY_CUSTOM0 + i);
+				array_len = custom_array.size() / factor_reciprocal;
```

Comment:
I know it's probably not a problem, but is there any chance ``custom_array.size()`` could return 0?

Reply:
Well, `ARRAY_CUSTOM0` can be any size, but if it’s 0, then the code should just behave the same way it does without my changes.

Reply:
The thing is, in C++, dividing by 0 is undefined. So there is a chance that doing this could introduce garbage into the variable that can screw things up. Having a check before the division to make sure that it's not 0 and bail if it is will prevent the garbage from happening.

Reply:
I’m not dividing by `custom_array.size()`, though. `custom_array.size()` is the dividend in this case. I’m dividing by `factor_reciprocal` which shouldn’t ever be 0.

Reply:
I used the wrong verbage, but the problem still remains. It's undefined behavior in C++, which could cause garbage values, even if the dividend is 0.

The code and the idea itself is good, I just feel like having a ``custom_array.size() != 0`` check before dividing would fix a potential problem source from undefined behavior and the dark magic that is the C++ compilers.

Reply:
> I used the wrong verbage, but the problem still remains. It's undefined behavior in C++, which could cause garbage values, even if the dividend is 0.
> 
> The code and the idea itself is good, I just feel like having a `custom_array.size() != 0` check before dividing would fix a potential problem source from undefined behavior and the dark magic that is the C++ compilers.

Dividing 0 by any integer other than 0 is well defined behaviour in C++ or any other language for that matter.
Checking the dividend for 0 would be pointless.

Reply:
After educating myself farther, I was wrong. I'm sorry about that.